examples/taxi-ride-akka/README.md

```diff
@@ -0,0 +1,95 @@
+by Cloudflow (but can be easily added manually).
+
+Main flow of Flink implementation is very elegant and rasy to read   
+````
+      val rides: DataStream[TaxiRide] =
+        readStream(inTaxiRide)
+          .filter { ride ⇒
+            ride.isStart.booleanValue
+          }
+          .keyBy("rideId")
+
+      val fares: DataStream[TaxiFare] =
+        readStream(inTaxiFare)
+          .keyBy("rideId")
+
+      val processed: DataStream[TaxiRideFare] =
+        rides
+          .connect(fares)
+          .flatMap(new EnrichmentFunction)
+
+      writeStream(out, processed)
+````
+
+Akka implementation provides the same functionality, with the following code:
+````
+      val rides = shardedSourceWithCommittableContext(inTaxiRide, entity).via(ridesFlow)
+
+      val fares = shardedSourceWithCommittableContext(inTaxiFare, entity).via(faresFlow)
+
+      Merger.source(Seq(rides, fares)).to(committableSink(out))
+````
+Note that here there is no explicit `keyBy` (we will discuss it later). 
+Unlike Flink implementation, where streams are connected before invoking, in the case of
+Akka, we process each stream independantly and then merge results afterwards.
+Finally individual flow looks like following:
+
+````
+      FlowWithCommittableContext[TaxiRide]
+        .mapAsync(1)(msg ⇒ {
+          val actor = sharding.entityRefFor(typeKey, msg.rideId.toString)
+          actor.ask[Option[TaxiRideFare]](ref => ProcessRide(ref, msg))
+        }).collect{ case Some(v) => v }
+````
+Here, `keyBy` is implemented through actor selection - we are using`rideId` is the actor ID.
+So the implementation provides the same functionality (except for automatic persistence, 
+which can be easily added), but with less user friendly syntax. Flink's implementation hides
+all of the sharding details by usage of `keyBy`.
+On another hand, FLink uses static topology for applications, which means that scaling requires
+application restarts, while in the case of Akka, adding additional instance can be done without 
+any restarts.
+
+The interesting feature of this example, is that the keys `rideID` do not belong to a fix set,
+but rather change all the time. As a result Akka implementation creates and deletes actors
+all the time. FLink alleviates this problem by using keys sharding - using a single implementation
+(actor, in Akka terms) for a set of keys.
+
+We can mimic the same functionality in Akka streams. To do this lets remember, that cloudflow
+is leveraging [Kafka aware sharding](https://doc.akka.io/docs/alpakka-kafka/current/cluster-sharding.html),
+which means that each instance processes a set of partitions from the topic. In the case of scaling,
+the partition as a whole can be moved to a different instance, but can not be split.
+This means that we can use actors based on partitions to avoid constant acters creation and deletion.
+
+To achieve this we need to make several changes to our implementation.
```

Comment:
It's probably better to just show one solution (the single message) and describe the reasoning for it when using Akka (colocating shard and consumer).

Reply:
i like options, sorry

Reply:
There are different use cases and both approaches have their sweet spots